src/usr/local/www/services_dhcp.php

```diff
@@ -1010,6 +1077,36 @@ function build_pooltable() {
 
 $form->add($section);
 
+//OMAPI
+$section = new Form_Section('OMAPI');
+
+$section->addInput(new Form_Input(
+	'omapi_port',
+	'OMAPI Port',
+	'text',
+	$pconfig['omapi_port']
+))->setAttribute('placeholder', 'OMAPI Port')->setHelp('Set the port that OMAPI will listen on. The default port is 7911, leave blank to disable.');
+
+$group = new Form_Group('OMAPI Key');
+
+$group->add(new Form_Input(
+	'omapi_key',
+	'OMAPI Key',
+	'text',
+	$pconfig['omapi_key']
+))->setAttribute('placeholder', 'OMAPI Key')->setHelp('Set the key used to secure the connection to the OMAPI endpoint. The minimum HMAC-MD5 key length is 256 bits.');
+
+$group->add(new Form_Checkbox(
+	'omapi_gen_key',
+	'',
+	'Generate New Key',
+	$pconfig['omapi_gen_key']
+))->setHelp('Key Bits: 256<br />Algorithm: HMAC-MD5');
```

Comment:
Might not be as viable here, but take a look at the captive portal voucher page and see how this is handled there. There is a button to generate the key, rather than creating it in the background when saving. Either way is fine, though.

Reply:
I took a look at this and I think I am going to stick with the current method.